src/js/components/modal.js

```diff
@@ -4,32 +4,41 @@ export class Modal {
     hideClassName = 'hide';
     modalHeight = '';
 
-    constructor(modalName, options) {
+    constructor(modalName, idShowModalButton, options) {
         this.modal = document.querySelector(`#${modalName}`);
+        this.buttonShowModal = document.querySelector(`#${idShowModalButton}`);
         this.options = options ?? {};
         this.closeButton = this.modal.querySelector('button');
         // this.#sayHello();
     }
 
     handleCloseButton() {
         this.closeButton.addEventListener('click', () => {
-            if(this.options.onCloseShowInfo) {
-                alert('CLOSE!')
-            } else {
-                this.modal.classList.add(this.hideClassName);
-                this.$bodyElement.classList.remove(this.bodyOverlayClassName);
-            }
+            this.$bodyElement.classList.remove(this.bodyOverlayClassName);
+            localStorage.setItem("hasBeenClickedButtonClose", "true");
+            this.modal.classList.add(this.hideClassName)
+            this.buttonShowModal.classList.remove(this.hideClassName);
+            this.handleShowModalButton();
         })
+    }
 
-        this.modalHeight = '1000px;';
+    handleShowModalButton(){
+        this.buttonShowModal.addEventListener('click',()=>{
+            this.init();
+        })
     }
 
     #sayHello() {
        console.log('hello');
     }
 
     init() {
-        this.$bodyElement.classList.add(this.bodyOverlayClassName);
+        this.modal.classList.remove(this.hideClassName)
+        if(localStorage["hasBeenClickedButtonClose"]!=="true"){
```

Comment:
to jest .. "okej" ale moglbys tutaj wykorzystac:

localStorage.getItem("hasBeenClickedButtonClose") !== null

czyli sprawdzac czy taki klucz istnieje, a nie jaka ma dokladnie wartosc, troszke ladniejszy zapis